server/neptune/workflows/internal/pr/revision/policy/handler.go

```diff
@@ -149,6 +151,35 @@ func (f *FailedPolicyHandler) fetchTeamMembers(ctx workflow.Context, repo gh.Rep
 	return listTeamMembersResponse.Members, err
 }
 
+func (f *FailedPolicyHandler) updateCheckStatus(ctx workflow.Context, roots map[string]revision.RootInfo, responses []terraform.Response, policies []activities.PolicySet) {
+	for _, response := range responses {
+		if containsFailingPolicy(response.ValidationResults, policies) {
+			continue
+		}
+		workflowState := response.WorkflowState
+		workflowState.Result.Status = state.CompleteWorkflowStatus
+		workflowState.Result.Reason = state.SuccessfulCompletionReason
```

Comment:
Kinda weird that we're mutating workflow state outside the workflow but there doesn't seem to be a better way. I'd say then that you should have a distinction.  Like this isn't really a `SuccessfulCompletionReason` The failure has just been approved, so can you capture that context here?